pkg/querier/queryrange/results_cache.go

```diff
@@ -133,12 +139,37 @@ func (s resultsCache) Do(ctx context.Context, r Request) (Response, error) {
 	return response, err
 }
 
+// shouldCacheResponse says whether the response should be cached or not.
+func shouldCacheResponse(r Response) bool {
+	if promResp, ok := r.(*PrometheusResponse); ok {
+		shouldNotCache := false
+		for _, hv := range promResp.Headers {
+			if hv == nil {
+				continue
+			}
+			if hv.Name != cachecontrolHeader {
+				continue
+			}
+			if hv.Values[0] == noCacheValue {
```

Comment:
This matches only for `Cache-Control: no-store`, while the `Cache-Control` header value could be a bit more articulated (ie. `Cache-Control: private, no-store`). Maybe we could make it a bit more robust:

1. Looking for `no-store` within the header value
2. Checking all header values (being a list)

Reply:
Good point, added a check for all header's values :smile: